QuickNoteVault.API/QuickNoteVault.DAL/UOW/UnitOfWork.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Microsoft.EntityFrameworkCore;
+using QuickNoteVault.DAL.Repository;
+
+namespace QuickNoteVault.DAL.UOW
+{
+    public class UnitOfWork : IUnitOfWork
+    {
+        private readonly DbContext _context;
+        private readonly Dictionary<Type, object> _repositories = new Dictionary<Type, object>();
+        private bool _disposed = false;
+
+        public UnitOfWork(DbContext context)
+        {
+            _context = context;
+        }
+
+        public IRepository<T> GetRepository<T>()
+            where T : class
+        {
+            var type = typeof(T);
+            if (!_repositories.ContainsKey(type))
+            {
+                var repositoryType = typeof(Repository<>).MakeGenericType(type);
+                var repositoryInstance = Activator.CreateInstance(repositoryType, _context);
```

Comment:
In this case just use `new Repository<T>(_context);`, rewrite method, make it simple